Assets/LeapMotionModules/InteractionEngine/Scripts/ControllerImplementations/HeuristicGrabClassifier.cs

```diff
@@ -0,0 +1,142 @@
+﻿using UnityEngine;
+using System;
+using System.Collections.Generic;
+using Leap.Unity.RuntimeGizmos;
+
+#if UNITY_EDITOR
+using UnityEditor;
+#endif
+
+namespace Leap.Unity.Interaction {
+  public class HeuristicGrabClassifier {
+    Collider[] collidingCandidates = new Collider[10];
+    public InteractionManager _manager;
+    private Hand _hand;
+
+    Dictionary<IInteractionBehaviour, GrabClassifier> leftGrabClassifiers = new Dictionary<IInteractionBehaviour, GrabClassifier>();
+    Dictionary<IInteractionBehaviour, GrabClassifier> rightGrabClassifiers = new Dictionary<IInteractionBehaviour, GrabClassifier>();
+
+    public HeuristicGrabClassifier(InteractionManager manager) {
+      _manager = manager;
+    }
+
+    //Grab Classifier Logic
+    public void UpdateClassifier(GrabClassifier classifier) {
+      classifier.warpTrans = Matrix4x4.TRS(classifier.warper.RigidbodyPosition, classifier.warper.RigidbodyRotation, classifier.transform.localScale);
+      
+      //For each probe (fingertip)
+      for (int j = 0; j < classifier.probes.Length; j++) {
+        //Calculate how extended the finger is
+        float tempCurl = Vector3.Dot(_hand.Fingers[j].Direction.ToVector3(), (j != 0) ? _hand.Direction.ToVector3() : (_hand.IsLeft ? 1f : -1f) * _hand.Basis.xBasis.ToVector3());
+
+        //Determine if this probe is intersecting an object
+        Array.Clear(collidingCandidates, 0, 10);
+        Physics.OverlapSphereNonAlloc(_hand.Fingers[j].TipPosition.ToVector3(), j == 0 ? 0.015f : 0.01f, collidingCandidates);
+        bool collidingWithObject = false;
+        foreach(Collider col in collidingCandidates) {
+          if (col!=null && col.attachedRigidbody != null) {
+            collidingWithObject = (col.attachedRigidbody == classifier.body) ? true : collidingWithObject;
```

Comment:
```
if (col!=null && col.attachedRigidbody != null && col.attachedRigidbody == classifier.body) {
  collidingWithObject = true;
  break;
}
```
This change allows you to break out of the loop early, since all you need to know is if at least one obj is colliding.